docs/topics/js/js-debugging.md

```diff
@@ -0,0 +1,99 @@
+   Depending on your browser, these logs can reference the Kotlin source files and lines they come from:
+
+   ![Chrome DevTools console](devtools-console.png)
+   {width="600"}
+
+4. Click the file reference on the right to navigate to the corresponding line of code.
+   Alternatively, you can manually switch to the **Sources** tab, and find the file you need in the file tree. Navigating
+   to the Kotlin file shows you the regular Kotlin code (as opposed to minified JavaScript):
+
+   ![Debugging in Chrome DevTools](devtools-sources.png)
+   {width="600"}
+
+You can now start debugging the program. For example, set a breakpoint by clicking on one of the line numbers.
+The developer tools even support setting breakpoints within a statement. As with regular JavaScript code, any set
+breakpoints will persist across page reloads. This also makes it possible to debug Kotlin's main method which is executed
+when the script is loaded for the first time.
+
+## Debug in the IDE
+
+[IntelliJ IDEA Ultimate](https://www.jetbrains.com/idea/) provides a powerful set of tools for debugging code during the development.
+
+For debugging Kotlin/JS in IntelliJ IDEA, you'll need a **Debug JavaScript** debug configuration. To add such a configuration:
+
+1. Go to **Run | Edit configurations**.
+2. Click **+** and select **JavaScript debug**.
+3. Specify the configuration **Name** and provide the **URL** on which the project runs (`http://localhost:8080` by default).
+
+   ![JavaScript debug configuration](debug-config.png)
+   {width="600"}
+
+   Learn more about [setting up JavaScript debug configurations](https://www.jetbrains.com/help/idea/configuring-javascript-debugger.html).
+4. Save the configuration.
+
+Now you're ready to debug your project:
+
+1. Run the project by calling one of the available `*Run` Gradle tasks, for example, `browserDevelopmentRun` or
+   `jsBrowserDevelopmentRun` in a multiplatform project.
+   Learn more in [this tutorial](running-kotlin-js.md#run-the-browser-target).
+2. Start the debugging session by running the JavaScript debug configuration you've created previously:
+
+   ![JavaScript debug configuration](debug-config-run.png)
+   {width="600"}
+
+3. You can see the console output of your program in the **Debug** window in IntelliJ IDEA. The output items reference the
+   Kotlin source files and lines they come from:
+
+   ![JavaScript debug output in IDE](ide-console-output.png)
+   {width="600"}
+
+4. Click the file reference on the right to navigate to the corresponding line of code.
+
+You can now start debugging the program using the whole set of tools that the IDE offers: breakpoints, stepping, expression
+evaluation, [coroutine debugger](debug-coroutines-with-idea.md) and more. Learn more about [debugging in IntelliJ IDEA](https://www.jetbrains.com/help/idea/debugging-code.html).
+
+## Debug in Node.js
+
+If your projects targets Node.js, you can debug it right in this runtime.
```

Comment:
```suggestion
If your projects target Node.js, you can debug them right in this runtime.
```

Reply:
Maybe it's better to rephrase in the singular.

Reply:
Yep, I'd use singular here.